x-pack/plugins/remote_clusters/server/lib/serialize_cluster.js

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export function serializeCluster(name, esClusterObject) {
```

Comment:
I guess the terms serialization and deserialization are fairly arbitrary in this context, but FWIW in rollups I considered this type of conversion to be "deserialization": https://github.com/elastic/kibana/blob/master/x-pack/plugins/rollup/public/crud_app/services/jobs.js#L79

Reply:
the context does make the naming a bit confusing - here I was thinking along the lines of "serialize this for UI consumption", then "deserialize UI object back for ES consumption"

I'm working on the deserialization part now, so will think a little bit more about which naming makes more sense. if swapping them makes more sense, will include that in the next PR!

Reply:
Sounds good! The reason I chose serialization for ES consumption was because Wikipedia defines [serialization](https://en.wikipedia.org/wiki/Serialization) as "the process of translating data structures or object state into a format that can be stored or transmitted and then reconstructed later" which seemed to have some similarities with what I was doing (formatting the object for transmitting via a request and then storing it in ES). But I'm open to reversing terminology or changing it entirely for something which makes more sense.